components/org.wso2.carbon.identity.api.server.api.resource/org.wso2.carbon.identity.api.server.api.resource.v1/src/main/java/org/wso2/carbon/identity/api/server/api/resource/v1/core/ServerAPIResourceManagementService.java

```diff
@@ -0,0 +1,475 @@
+            List<String> removedScopeNames = new ArrayList<>();
+
+            APIResource apiResource = new APIResource();
+            apiResource.setName(displayName);
+            apiResource.setId(apiResourceID);
+            apiResource.setDescription(description);
+            APIResourceManagementServiceHolder.getApiResourceManager().updateAPIResource(apiResource, addedScopes,
+                    removedScopeNames, ContextLoader.getTenantDomainFromContext());
+        } catch (APIResourceMgtException e) {
+            throw APIResourceMgtEndpointUtil.handleAPIResourceMgtException(e);
+        }
+    }
+
+    /**
+     * Delete API Resource by ID.
+     *
+     * @param apiResourceID API Resource ID.
+     */
+    public void deleteAPIResource(String apiResourceID) {
+
+        try {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Deleting API Resource with ID: " + apiResourceID);
+            }
+            APIResourceManagementServiceHolder.getApiResourceManager().deleteAPIResourceById(apiResourceID,
+                    ContextLoader.getTenantDomainFromContext());
+        } catch (APIResourceMgtException e) {
+            throw APIResourceMgtEndpointUtil.handleAPIResourceMgtException(e);
+        }
+    }
+
+    /**
+     * Get scopes by API ID.
+     *
+     * @param apiResourceId API resource id.
+     * @return List of scopes.
+     */
+    public List<Scope> getScopesByAPIId(String apiResourceId) {
+
+        try {
+            APIResource apiResource = APIResourceManagementServiceHolder.getApiResourceManager()
+                    .getAPIResourceById(apiResourceId, ContextLoader.getTenantDomainFromContext());
+            if (apiResource == null) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("API Resource not found for ID: " + apiResourceId);
+                }
+                throw APIResourceMgtEndpointUtil.handleException(Response.Status.NOT_FOUND,
+                        APIResourceMgtEndpointConstants.ErrorMessage.ERROR_CODE_API_RESOURCE_NOT_FOUND, apiResourceId);
+            }
+            return apiResource.getScopes();
+        } catch (APIResourceMgtException e) {
+            throw APIResourceMgtEndpointUtil.handleAPIResourceMgtException(e);
+        }
+    }
+
+    /**
+     * Put scopes by API ID. This replaces existing scopes.
+     *
+     * @param apiResourceId       API resource id.
+     * @param scopeCreationModels Scope creation models.
+     */
+    public void putScopesByAPIId(String apiResourceId, List<ScopeCreationModel> scopeCreationModels) {
+
+        try {
+            APIResource apiResource = APIResourceManagementServiceHolder.getApiResourceManager()
+                    .getAPIResourceById(apiResourceId, ContextLoader.getTenantDomainFromContext());
+            if (apiResource == null) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("API Resource not found for ID: " + apiResourceId);
+                }
```

Comment:
unnecessary debug log.

Reply:
fix the other places also.

Reply:
Fixed with https://github.com/wso2/identity-api-server/pull/473/commits/5c51c9eff45d1c55a16c7858a056b5a9eed2ce86